core/common/src/main/scala/org/typelevel/otel4s/Attributes.scala

```diff
@@ -152,6 +170,34 @@ sealed trait Attributes
 
   override def toString: String =
     Show[Attributes].show(this)
+
+  private class TypedMap
```

Comment:
Should we keep the `TypedMap` in the companion object?
```scala
def asTypedMap: Map[AttributeKey[_], Attribute[_]] = new TypedMap(asMap)
```

Then we don't need `self`.

Reply:
Let's keep it as is.